### DISCUSSION TITLE
We need more maintainers

### DISCUSSION THREAD

**Discussion by drbacke** [Akkudoktor-EOS/EOS#97] [Announcements]
We need more maintainers so that the pr can be processed more quickly. Who would be interested? 
Many thanks in advance

**Comment by fortrieb** [comment]
I'm interested on maintaining this project. Are there any requirements needs to be fulfilled before?

**Comment by drbacke** [comment]
I would say, since we are still a very small group and I wanted to ask you anyway. Don't make it too complicated. I've signed you up as a maintainer.
As soon as we have a good core group here, I would promote the project more and get more people involved. But it's still too early for that

**Comment by michaelosthege** [comment]
I'm willing to contribute with code review, refactoring and related maintenance work for while.
However, I can't commit to maintaining long-term because I don't have the perspective of using the system myself in the near future (lacking PV, battery, e-car...).

**Reply by drbacke** [reply]
Thank you very much. You are a maintainer now😊

**Comment by sandboxcode** [comment]
I am also really interested to join as a maintainer. I lead the IT department of a small pv and charging infrastructure installer in Hamburg. We are also working on other software tools that are in private development right now, but we plan to open source it in the future. 
We do also have multiple sites where the owners (clients and or CEO / CTO) allow my team to play around and test around.
Might be really helpful when we want to apply it in the "wild".
Happy to join!

**Comment by BerndCzech** [comment]
I would be open for maintaining. However to not loose focus I would step back in favor of having a little more focus at the top.

**Reply by drbacke** [reply]
You are a maintainer now.

**Comment by michaelosthege** [comment]
@drbacke can you connect all current maintainers? Maybe in a Discord video call or something like that.

It would help tremendously to have a short exchange about priorities, expectations and aligning acceptance criteria for PRs.

**Comment by drbacke** [comment]
Sorry I don't have time to organize a meeting, also 1 hour of meeting is very difficult for me. 
We should do this asynchronous in some kind of chat or in discussions. 
Or it will be in some weeks.

**Comment by BerndCzech** [comment]
@michaelosthege if you could write a strategy proposal we can refer to and discuss, we might do a first step.

**Reply by michaelosthege** [reply]
Yes, here we go: #187

**Comment by Krulle86** [comment]
Next supporter for your project.
What tasks are pending? Is there a wish list?

**Reply by fortrieb** [reply]
Unit tests are priority number one at the moment, to not adding more bugs while merge PR into the project.

**Comment by NormannK** [comment]
could the maintainers please push some PR? I do understand some might be complicated/controversial but some are super easy and simple bug fixes.

**Reply by drbacke** [reply]
You are a maintainer now.

**Comment by fracbar** [comment]
Someone already working on a container release? If not, i can provide a simple workflow script, which will push nightlies (and release) images to the public docker repo. This can make life easier to install/test it out.

**Comment by eckelj** [comment]
I am also interested in supporting and maintaining the project. are there active transitions that need support? just arrived here on the project, so excuse this maybe naive question.

**Comment by SREbuilt** [comment]
My recommendation are to 
(1) protect the main branch by 
- automated integration tests,
- enable check-in policies in the project config,
- (in the future) have automated deployment tests,
and not only rely on reviewers for their manual review (having tests will free a bit of their capacity).

The goal of this is that the code in the main branch will always be operational.
As of today (2024-11-10) the main branch contains code that is not executable: Several error types are thrown of categories modules are not found (in akkudoktoreos subfolder), and functional errors like exceptions are happening once server starts.

(2) have a branching strategy in a way that all the development (fixing bugs and implement new features and improvements, like refactoring) to only happen in feature branches (of atomic scope/single purpose and short lifetime).
This will help to keep a better overview of what's happening by whom (to avoid duplicate work).

(3) Setup a Project for this repo: 
This will have a simple Kanban board that helps to give an overview and to coordinate the contributions amongst themselves.

@drbacke : in case further maintainers/contributors are welcome, let me be one. I've a Site Reliability Engineering background (with focus on solution observability, increase efficiency (reduce toil and increase automation) ...

**Reply by drbacke** [reply]
These are very good suggestions, but the biggest problem at the moment is that I'm very busy and therefore can't do the maintaining well. 
But I'd be happy to add you as a maintainer, that's very important, that's what's currently lacking the most.

**Reply by michaelosthege** [reply]
What's currently lacking most is coordinatation among maintainers and/or a call by you on who should make decisions with bigger impact.

For example, in https://github.com/Akkudoktor-EOS/EOS/discussions/187 I proposed a strategy, and my PR https://github.com/Akkudoktor-EOS/EOS/pull/188 does a first step, but merging it would break several currently open PRs. I have maintainer permissions, but I don't want to make that decision alone.
Note that there's no easy way for us to see who the maintainers are.

**Reply by Krulle86** [reply]
I think we should use a communication platform: Discord or something like that.

**Reply by NormannK** [reply]
how about you setup a Discord as a start

**Reply by SREbuilt** [reply]
> These are very good suggestions, but the biggest problem at the moment is that I'm very busy and therefore can't do the maintaining well. But I'd be happy to add you as a maintainer, that's very important, that's what's currently lacking the most.

@drbacke :many thanks for onboarding me.
Can you please add a Kanban board to the Project tab? Seems only the repo owner can do that.
Thanks, 
Bernhard

**Reply by SREbuilt** [reply]
regarding Discord: 
Who ever has experience with that: go for it. Let's try this medium.
For sure we want to get aligned and make a couple of decisions...
I.e. we could 
- converge all existing PR and have them merged into main (to not loose any of the valuable contributions)
- not start any new PR
- do the major refactoring (that would break any existing PRs)
- and then start dev in feature and bug-fixing branches only
- have the main branch protected by check-in policies and automated CI-tests + approvals of 1 or 2 reviewers.

**Reply by NormannK** [reply]
yes I agree. if you would take a look into my PR's it would be great and most open would be gone.
Id rather not merge them myself. 
Id love to have the battery Refactoring in #130 also included after that most stuff should be at least english

**Reply by SREbuilt** [reply]
@drbacke : to add a Kanban board (sorry for nagging):
- click on your Profile picture (top right) > 
- Your Projects > 
- New Project > 
- Choose the Kanban board.
- then follow steps to link the project in the EOS repository, described here: https://docs.github.com/en/issues/planning-and-tracking-with-projects/managing-your-project/adding-your-project-to-a-repository


sources: 
- https://docs.github.com/en/issues/planning-and-tracking-with-projects/creating-projects/creating-a-project
- https://docs.github.com/en/issues/planning-and-tracking-with-projects/managing-your-project/adding-your-project-to-a-repository

Thanks

**Comment by Krulle86** [comment]
I've started an almost blank discord-server: 
https://discord.com/invite/ZpZEmZvm

Feel free to step in and improve the thing.

**Comment by JanMarcoRuizdeVargas** [comment]
Hi there, I'm also potentially interested in supporting this project. My knowledge is mostly in machine learning. Where could I contribute?

**Reply by b0661** [reply]
Hi @JanMarcoRuizdeVargas,

Thank you for stepping in! We truly appreciate any contribution.

At the moment, we are not using machine learning. That said, predictions (e.g., PV forecast, load forecast) could potentially be improved with ML and historical data.

**Comment by DerOetzi** [comment]
Hi,

I'd like to briefly introduce myself. I've been developing open-source projects for smart home automation since 2012, initially for ioBroker and openHAB. In recent years, I've mainly focused on my own open-source projects:

- LearningHouseService: A RESTful API for machine learning-based smart home automation without hardcoded conditional rules.
- Since 2023: solaredge2mqtt, a tool to locally monitor and forecast my PV system using Modbus, MQTT, Home Assistant auto-discovery and scikit-learn for machine-learning based forecast
- Since 2024: @deroetzi/node-red-contrib-smarthome-helper, a set of easy-to-use Node-RED helper nodes for smart home setups.

For some months now, I've been planning to rewrite my LearningHouseService into a LearningHouse Energy Service for intelligent sector coupling. However, I recently discovered your open-source project and believe it might be a better use of my time and skills to contribute to your work rather than starting from scratch again.

What I can offer:

- 15+ years of experience as a software developer and architect
- Strong knowledge in: Python, Angular, Java, TypeScript, Docker
- Deep understanding of smart home automation
- Experienced with Python libraries like pymodbus, fastapi, scikit-learn, and pydantic
- A production forecast model that can be trained locally with system-specific data
- A framework for easily developing custom Node-RED nodes

If you're interested in a collaboration, I'd be happy to get in touch!

Best regards,
Johannes

**Reply by b0661** [reply]
Welcome, thank you for offering your help. Your experience and contribution is very much appreciated.

E.g. the production forecast model could become an EOS prediction for PV. You might have a look at the [Akkudoktor PV Forecast as an example](https://github.com/Akkudoktor-EOS/EOS/blob/main/src/akkudoktoreos/prediction/pvforecastakkudoktor.py).

The are a lot more area that also needs help (e.g. documentation,  a common framework for device simulations,, ...). Feel free to choose at your preference.

Cheers Bobby

**Reply by DerOetzi** [reply]
Okay, I have to finish an extension for my solaredge2mqtt. I estimate round about 2 weeks for this. After this I can start to contribute. 

What we will need is access to some hourly weather forecast and the possibility to get the production data of the last hour to combine it with weather forecast for training.

For weather forecast i currently use openweathermap but every other weather api giving hourly forecast will be good. 

Is there any other communication platform to get in touch more? I have seen a discord link above, but the link didn't work.

**Comment by SchaafAlexander** [comment]
Hi all,

I would like to contribute as well by not only making suggestions to the README ;-)
Background of +10 years in data visualization, modeling, pipelines etc.

I have integrated EOS on my server fed with the necessary data and getting the first solid predictions.
Next steps:
 - to fully understand the configuration file as the PV forecast is still off.
- give EOS control over the PV/charger
 
In regards to contribution I would start small with the vizes, as they trigger my "inner Monk" ;-)

Thank you very much & have a great day

**Reply by drbacke** [reply]
Hey Alexander,
this Sounds great. Maybe I could invite you, to directly communicate with us.
If you are interested, contact me via info@akkudoktor.net
